test/src/hls-adapter.spec.js

```diff
@@ -483,6 +484,109 @@ describe('HlsAdapter Instance - change media', function() {
       });
     });
   });
+
+  it('should fire FRAG_LOADED', done => {
+    try {
+      hlsAdapterInstance = HlsAdapter.createAdapter(video, source1, config);
+      hlsAdapterInstance.addEventListener(EventType.FRAG_LOADED, event => {
+        event.payload.miliSeconds.should.exist;
+        event.payload.bytes.should.exist;
+        event.payload.url.should.not.be.empty;
+        done();
+      });
+      hlsAdapterInstance.load();
+    } catch (e) {
+      done(e);
+    }
+  });
+
+  it('should fire MANIFEST_LOADED', done => {
+    try {
+      hlsAdapterInstance = HlsAdapter.createAdapter(video, source1, config);
+      hlsAdapterInstance.addEventListener(EventType.MANIFEST_LOADED, event => {
+        event.payload.miliSeconds.should.exist;
+        done();
+      });
+    } catch (e) {
+      done(e);
+    }
+    hlsAdapterInstance.load();
+  });
+
+  it('should check targetBuffer in VOD far from end of stream', done => {
+    try {
+      hlsAdapterInstance = HlsAdapter.createAdapter(video, source1, config);
+      video.addEventListener(EventType.PLAYING, () => {
+        hlsAdapterInstance.targetBuffer.should.equal(
+          hlsAdapterInstance._hls.config.maxMaxBufferLength + hlsAdapterInstance._getLevelDetails().targetduration
+        );
+        done();
+      });
+      hlsAdapterInstance.load().then(() => {
+        video.play();
+      });
+    } catch (e) {
+      done(e);
+    }
+  });
+
+  it('should check targetBuffer in VOD close to end of stream (time left to end of stream is less than targetBuffer)', done => {
+    try {
+      hlsAdapterInstance = HlsAdapter.createAdapter(video, source1, config);
+      video.addEventListener(EventType.PLAYING, () => {
+        video.currentTime = video.duration - 1;
+        video.addEventListener(EventType.SEEKED, () => {
+          hlsAdapterInstance.targetBuffer.should.equal(video.duration - video.currentTime);
+          done();
+        });
+      });
+      hlsAdapterInstance.load().then(() => {
+        video.play();
+      });
+    } catch (e) {
+      done(e);
+    }
+  });
+
+  it('should check targetBuffer in LIVE', done => {
+    try {
+      hlsAdapterInstance = HlsAdapter.createAdapter(video, liveSource, config);
+      video.addEventListener(EventType.PLAYING, () => {
+        hlsAdapterInstance._hls.config.maxMaxBufferLength = 120;
```

Comment:
why do you set the `maxMaxBufferLength` config after playing? we don't set HLSJS config on the fly on adapters so I think it'll be better to set the value before the adapter creation.

Reply:
done

Reply:
done